analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs

```diff
@@ -126,4 +126,11 @@ class ImplementsMarker: IMarker { }          // Compliant - implements a marker
     struct EmptyStruct { }                       // Compliant - this rule only deals with classes
 
     enum EmptyEnum { }                           // Compliant - this rule only deals with classes
+
+    public class BaseWithNonPublicConstructor
+    {
+        protected BaseWithNonPublicConstructor() { }
+    }
+
+    public class WidensTheConstructorVisibility : BaseWithNonPublicConstructor { }  // Compliant
```

Comment:
Can you add more UTs? like different access modifiers (public/private/internal) and primary constructors (if not present already)